### DISCUSSION TITLE
Help highlighting builtins/keywords for my grammar

### DISCUSSION THREAD

**Discussion by FoamScience** [tree-sitter/tree-sitter#1490] [Q&A] [answered]
Hi,

I have a test file for my grammar:
```
FoamFile {
    version 2.0;
}
```

Which parses like this:
```
(source_file [0, 0] - [3, 0]
  (dict [0, 0] - [2, 1]
    key: (identifier [0, 0] - [0, 8])
    dict_body: (key_value [1, 4] - [1, 16]
      keyword: (identifier [1, 4] - [1, 11])
      value: (number_literal [1, 12] - [1, 15]))))
```

With `tree-sitter-cli` compiled from master, I'm trying to highlight it with this scm file:
```
(dict key: (identifier) @type)
(number_literal) @number

"version" @type
```

But it errors out, saying "**Query error at 4:2. Invalid node type version**".
Other highlighters can use strings for keywords just fine; but I can't. Anyone knows what's wrong?

PS: when I remove that line, everything works as expected. Any help is much appreciated!

**Comment by ahelwer** [comment, marked as answer]
You can only use strings in queries if that string is an explicit token in your language, not matched by a regex.

**Reply by FoamScience** [reply]
Oh; cool! Thanks, will have to do some changes to the parser then. Thanks again for the lightening-speed response!
